kubernetes_test.go

```diff
@@ -429,6 +430,41 @@ func TestKubernetesClusters_GetCredentials(t *testing.T) {
 }`
 	mux.HandleFunc("/v2/kubernetes/clusters/deadbeef-dead-4aa5-beef-deadbeef347d/credentials", func(w http.ResponseWriter, r *http.Request) {
 		testMethod(t, r, http.MethodGet)
+		if len(r.URL.Query()) > 0 {
+			t.Error("Request should not have Url params")
+		}
+		fmt.Fprint(w, jBlob)
+	})
+	got, _, err := kubeSvc.GetCredentials(ctx, "deadbeef-dead-4aa5-beef-deadbeef347d", &KubernetesClusterCredentialsGetRequest{})
+	require.NoError(t, err)
+	require.Equal(t, want, got)
+}
+
+func TestKubernetesClusters_GetCredentials_WithExpirySeconds(t *testing.T) {
+	setup()
+	defer teardown()
+
+	kubeSvc := client.Kubernetes
+	timestamp, err := time.Parse(time.RFC3339, "2014-11-12T11:45:26.371Z")
+	require.NoError(t, err)
+	want := &KubernetesClusterCredentials{
+		Token:     "secret",
+		ExpiresAt: timestamp,
+	}
+	jBlob := `
+{
+	"token": "secret",
+	"expires_at": "2014-11-12T11:45:26.371Z"
+}`
+	mux.HandleFunc("/v2/kubernetes/clusters/deadbeef-dead-4aa5-beef-deadbeef347d/credentials", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodGet)
+		expirySeconds, ok := r.URL.Query()["expiry_seconds"]
+		if !ok {
+			t.Error("Url param 'expiry_seconds' is missing")
+		}
+		if len(expirySeconds) != 1 || expirySeconds[0] != "3600" {
```

Comment:
If we wanted these checks to be implemented using testify, we could do:

```go
assert.Len(t, expirySeconds, 1)
assert.Contains(t, expirySeconds, "3600")
```

Reply:
fixed